src/Microsoft.AspNetCore.SignalR.Common/Internal/Formatters/BinaryMessageParser.cs

```diff
@@ -70,5 +71,16 @@ public static bool TryParseMessage(ref ReadOnlyMemory<byte> buffer, out ReadOnly
             buffer = buffer.Slice(numBytes + (int)length);
             return true;
         }
+
+        private static ReadOnlySpan<byte> GetSpan(in ReadOnlySequence<byte> lengthPrefixBuffer)
+        {
+            if (lengthPrefixBuffer.IsSingleSegment)
+            {
+                return lengthPrefixBuffer.First.Span;
+            }
+            
+            // Should be rare
+            return lengthPrefixBuffer.ToArray();
```

Comment:
We could stackalloc but meh.

Reply:
What about inlining this logic up, and moving the parsing to a helper method that takes `Span<byte>`?

```csharp
int length;
if(buffer.First.Span.Length >= 5) {
    length = ParseLengthPrefix(buffer.First.Span.Slice(0, 5));
} else {
    Span<byte> stackBuffer = stackalloc byte[5];
    buffer.Slice(0, 5).CopyTo(stackBuffer);
    length = ParseLengthPrefix(stackBuffer);
}
```

Reply:
Yea, that works.